post-til-virksomhet/src/main/java/no/difi/meldingsutveksling/ptv/CorrespondenceAgencyMessageFactory.java

```diff
@@ -0,0 +1,218 @@
+        no.altinn.schemas.services.serviceengine.notification._2009._10.ObjectFactory notificationFactory = new no.altinn.schemas.services.serviceengine.notification._2009._10.ObjectFactory();
+        notification.setFromAddress(notificationFactory.createNotification2009FromAddress("no-reply@altinn.no"));
+        // The date and time the notification should be sent
+        notification.setShipmentDateTime(toXmlGregorianCalendar(DateTime.now().plusMinutes(5)));
+        // Language code of the notification
+        notification.setLanguageCode(notificationFactory.createNotification2009LanguageCode("1044"));
+        // Notification type
+        notification.setNotificationType(notificationFactory.createNotification2009NotificationType("offentlig_etat"));
+
+        TextTokenSubstitutionBEList tokens = new TextTokenSubstitutionBEList();
+        // $envelope/sender/orgnr - trenger en navnmapping TODO: Skal hentes fra SR basert paa orgnr
+        tokens.getTextToken().add(createTextToken(0, msg.getEnvelope().getSender().getName()));
+        // Message area, based on ServiceEdition
+        tokens.getTextToken().add(createTextToken(1, serviceEditionMapping.get(Integer.valueOf(serviceEdition))));
+        // Name of the message recipient TODO: Skal hentes fra SR basert paa orgnr
+        tokens.getTextToken().add(createTextToken(2, msg.getEnvelope().getReceiver().getName()));
+
+        notification.setTextTokens(notificationFactory.createNotification2009TextTokens(tokens));
+
+        JAXBElement<ReceiverEndPointBEList> receiverEndpoints = createReceiverEndPoint();
+        notification.setReceiverEndPoints(receiverEndpoints);
+
+        // Should the user be allowed to forward the message from portal
+        correspondence.setAllowForwarding(objectFactory.createMyInsertCorrespondenceV2AllowForwarding(false));
+        // Name of the message sender, always "Avsender"
+        correspondence.setMessageSender(objectFactory.createMyInsertCorrespondenceV2MessageSender("Avsender"));
+
+        notifications.getNotification().add(notification);
+        correspondence.setNotifications(objectFactory.createMyInsertCorrespondenceV2Notifications(notifications));
+
+        no.altinn.services.serviceengine.correspondence._2009._10.ObjectFactory correspondenceObjectFactory = new no.altinn.services.serviceengine.correspondence._2009._10.ObjectFactory();
+        final InsertCorrespondenceV2 myInsertCorrespondenceV2 = correspondenceObjectFactory.createInsertCorrespondenceV2();
+        myInsertCorrespondenceV2.setCorrespondence(correspondence);
+        myInsertCorrespondenceV2.setSystemUserCode(systemUserCode);
+        // Reference set by the message sender
+        myInsertCorrespondenceV2.setExternalShipmentReference(msg.getEnvelope().getConversationId());
+
+        return myInsertCorrespondenceV2;
+    }
+
+    private static String queryPayload(PutMessageRequestWrapper message, String xpath) {
```

Comment:
Jeg ville vurdert å trekke denne ut i en klasse som kan brukes her og av JournalPostId.from() for å unngå duplisering av denne koden. Så hatt unit tester for denne biten. Fordi erfaringsmessig så er alt som har med Payload en typisk ting som kan feile etter hvert som pilotvirksomhetene tester siden vi ikke har XSD for payload og payload er inni et <any> element.